tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```diff
@@ -0,0 +1,384 @@
+ * limitations under the License.
+ */
+
+import java.io.*;
+import java.util.ArrayList;
+import org.slf4j.Logger;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.Bucket;
+import com.amazonaws.services.s3.model.CannedAccessControlList;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.ObjectListing;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.slf4j.LoggerFactory.getLogger;
+import static java.nio.charset.Charset.defaultCharset;
+
+/**
+ * S3 client based on the Amazon AWS client API
+ */
+public class S3Client
+{
+    private static final Logger logger = getLogger(S3Client.class);
+
+    public static final String CONF_S3_AUTH_KEY_ID = "s3.auth.keyid";
+    public static final String CONF_S3_AUTH_SECRET = "s3.auth.secret";
+    public static final String CONF_S3_BUCKET_NAME = "s3.bucketname";
+
+    private final String bucketName;
+    private final AmazonS3 s3client;
+
+    /**
+     * Constructor
+     *
+     * @param keyId         AWS access key ID
+     * @param secret        AWS secret access key
+     * @param bucketName    S3 bucket to work in
+     */
+    public S3Client(
+            @Named(CONF_S3_AUTH_KEY_ID) String keyId,
+            @Named(CONF_S3_AUTH_SECRET) String secret,
+            @Named(CONF_S3_BUCKET_NAME) String bucketName)
+    {
+        // Check null values and set object variables
+        checkNotNull(keyId, "aws api access key id is null");
+        checkNotNull(secret, "aws api secret access key is null");
+        this.bucketName = checkNotNull(bucketName, "s3 bucket name is null");
+
+        // Initialize AWS SDK credentials
+        BasicAWSCredentials credentials = null;
+        try {
+            credentials = new BasicAWSCredentials(keyId, secret);
+        } catch (Exception e) {
+            throw new AmazonClientException("Cannot initialize credentials", e);
+        }
+
+        // Create S3 client
+        client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                .build();
+    }
+
+    /**
```

Comment:
sugg: drop javadoc

This applies to all javadocs here. Try to capture the actual responsibility of the method in its name.